### PR TITLE
feat(protocol): Genesis / DeployOnL1 with new proxy upgrades

### PR DESCRIPTION
**Pull request by cyberhorsey**
@dantaik @adaki2004  @davidtaikocha  this makes both `GenerateGenesis` and `DeployOnL1` work again, however, TaikoL1 does not work without removing one line of code. It is too big. It is at exactly the kb size limit. So TaikoL1 needs to be refactored.

Here is a screenshot of `pnpm run deploy:foundry` with removign one line of code from TaikoL1.

Screenshot from 2023-11-26 22-56-19

However, it will fail in this CICD, until refactored.

### CODE REVIEW

File: packages/protocol/genesis/GenerateGenesis.g.sol [hunk 1]
```diff
@@ -1,33 +1,30 @@
...
+import "forge-std/Test.sol";
+import "../contracts/common/AddressManager.sol";
+import "../contracts/common/AddressResolver.sol";
+import "../contracts/common/EssentialContract.sol";
+import "../contracts/bridge/Bridge.sol";
+import "../contracts/tokenvault/ERC1155Vault.sol";
+import "../contracts/tokenvault/ERC20Vault.sol";
+import "../contracts/tokenvault/ERC721Vault.sol";
+import "../contracts/bridge/IBridge.sol";
+import "../contracts/common/OwnerUUPSUpgradable.sol";
```
Comment (dantaik) [inline packages/protocol/genesis/GenerateGenesis.g.sol:15]: maybe we can remove imports for 
- OwnerUUPSUpgradable
- AddressManager
- AddressResolver
- IBridge
as they are all indirectly imported by EssentialContract

File: packages/protocol/genesis/GenerateGenesis.g.sol [hunk 2]
```diff
@@ -77,70 +74,62 @@ contract TestGenerateGenesis is Test, AddressResolver {
     }
 
     function testAddressManager() public {
-        AddressManager addressManager =
+        AddressManager addressManagerProxy =
```
Comment (dantaik) [inline packages/protocol/genesis/GenerateGenesis.g.sol:77]: We should simplify the names buy removing "Proxy", "Singleton"... as all contracts are behnind proxy.

File: packages/protocol/utils/generate_genesis/taikoL2.ts [hunk 3]
```diff
@@ -76,7 +76,9 @@ export async function deployTaikoL2(
         // reading it using smock package.
         let storageLayoutName = contractName;
         if (contractConfig.isProxy) {
-            storageLayoutName = contractName.replace("Proxy", "");
+            storageLayoutName = contractName
```
Comment (dantaik) [inline packages/protocol/utils/generate_genesis/taikoL2.ts:79]: With the latest changes, we don't have `Proxy` or `Singleton` in the name

File: packages/protocol/genesis/GenerateGenesis.g.sol [hunk 4]
```diff
@@ -77,70 +74,62 @@ contract TestGenerateGenesis is Test, AddressResolver {
...
+        checkSavedAddress(addressManagerProxy, "SingletonERC721VaultProxy", "erc721_vault");
+        checkSavedAddress(addressManagerProxy, "SingletonERC1155VaultProxy", "erc1155_vault");
+        checkSavedAddress(addressManagerProxy, "SingletonSignalServiceProxy", "signal_service");
+
+        AddressManager addressManager =
+            AddressManager(getPredeployedContractAddress("AddressManagerForSingletons"));
         AddressManager newAddressManager = new AddressManager();
 
-        vm.startPrank(admin);
+        vm.startPrank(addressManager.owner());
```
Comment (dantaik) [inline packages/protocol/genesis/GenerateGenesis.g.sol:114]: For all L2 contracts on mainnet, the owner shall be our security council.

    address public constant MAINNET_SECURITY_COUNCIL = 0x7C50d60743D3FCe5a39FdbF687AFbAe5acFF49Fd;

**Comment by dantaik** [comment]
@cyberhorsey I made very small changes in this commit, let me know if the code is smaller.

https://github.com/taikoxyz/taiko-mono/pull/15265/commits

**Review by dantaik — CHANGES_REQUESTED** [review]

**Comment by davidtaikocha** [comment]
> @cyberhorsey I made very small changes in this commit, let me know if the code is smaller.
> 
> https://github.com/taikoxyz/taiko-mono/pull/15265/commits

Looks like this commit didn't fix the size issue:

image

**Review by dantaik — APPROVED** [review]

**Comment by cyberhorsey** [comment]
@dantaik after merging proxy base back into his branch, tests fail again on TaikoGovernor init function in deploy script. I'll take a look at it tonight.

**Comment by davidtaikocha** [comment]
> @dantaik after merging proxy base back into his branch, tests fail again on TaikoGovernor init function in deploy script. I'll take a look at it tonight.

i'm looking into it too

**Comment by cyberhorsey** [comment]
OK, looks good, I will merge now.

**Pull request merged**